src/coprocessor/dag/expr/builtin_encryption.rs

```diff
@@ -21,6 +25,62 @@ const SHA256: i64 = 256;
 const SHA384: i64 = 384;
 const SHA512: i64 = 512;
 
+/// Numerical identifiers for an OpenSSL object supported in TiKV.
+const NID: phf::Set<i32> = phf_set! {
+    418i32, // aes-128-ecb
+    419i32, // aes-128-cbc
+    420i32, // aes-128-ofb128
+    421i32, // aes-128-cfb128
+    422i32, // aes-192-ecb
+    423i32, // aes-192-cbc
+    424i32, // aes-192-ofb128
+    425i32, // aes-192-cfb128
+    426i32, // aes-256-ecb
+    427i32, // aes-256-cbc
+    428i32, // aes-256-ofb128
+    429i32, // aes-256-cfb128
+    650i32, // aes-128-cfb1
+    651i32, // aes-192-cfb1
+    652i32, // aes-256-cfb1
+    653i32, // aes-128-cfb8
+    654i32, // aes-192-cfb8
+    655i32, // aes-256-cfb8
+};
+
+/// Generate a sequence of random bytes in length of `len`
+/// TODO: upgrade crate `rand` to 0.6+
+/// with `rand` 0.6+, the code can be written:
+/// ```compile_fail
+/// Standard.sample_iter(&mut thread_rng()).take(len).collect()
+/// ```
+fn random_bytes(len: usize) -> Vec<u8> {
+    let mut rng = rand::thread_rng();
+    (0..len).map(|_| rng.gen()).collect()
```

Comment:
With all versions of `rand` from 0.3 to 0.6 you could simply write

```suggestion
    let mut bytes = vec![0u8; len];
    rng.fill_bytes(&mut bytes);
    bytes
```

This is much faster even considering the zero-initialization. With rand 0.6.5:

```
test bench_fill_bytes_long        ... bench:       1,068 ns/iter (+/- 103)
test bench_fill_bytes_short       ... bench:          45 ns/iter (+/- 6)
test bench_loop_and_collect_long  ... bench:       7,407 ns/iter (+/- 1,157)
test bench_loop_and_collect_short ... bench:          85 ns/iter (+/- 10)
```

With rand 0.3.14:

```
test bench_fill_bytes_long        ... bench:       3,834 ns/iter (+/- 208)
test bench_fill_bytes_short       ... bench:          59 ns/iter (+/- 2)
test bench_loop_and_collect_long  ... bench:      17,345 ns/iter (+/- 1,747)
test bench_loop_and_collect_short ... bench:         164 ns/iter (+/- 7)
```

<details><summary>Benchmark</summary>

```rust
#![feature(test)]

extern crate test;
extern crate rand;

use test::{Bencher, black_box};
use rand::{Rng, RngCore, thread_rng};

#[bench]
fn bench_loop_and_collect_long(b: &mut Bencher) {
	let mut rng = thread_rng();
	let len = 2048;
	b.iter(|| {
		black_box((0..black_box(len)).map(|_| rng.gen()).collect::<Vec<u8>>());
	});
}

#[bench]
fn bench_fill_bytes_long(b: &mut Bencher) {
        let mut rng = thread_rng();
        let len = 2048;
        b.iter(|| {
		let mut bytes = vec![0u8; black_box(len)];
		rng.fill_bytes(&mut bytes);
		black_box(bytes);
        });
}

#[bench]
fn bench_loop_and_collect_short(b: &mut Bencher) {
        let mut rng = thread_rng();
        let len = 16;
        b.iter(|| {
                black_box((0..black_box(len)).map(|_| rng.gen()).collect::<Vec<u8>>());
        });
}

#[bench]
fn bench_fill_bytes_short(b: &mut Bencher) {
        let mut rng = thread_rng();
        let len = 16;
        b.iter(|| {
                let mut bytes = vec![0u8; black_box(len)];
                rng.fill_bytes(&mut bytes);
                black_box(bytes);
        });
}
```

</details>